engine/sound/src/sound.cpp

```diff
@@ -518,6 +520,11 @@ namespace dmSound
 
     Result DeleteSoundData(HSoundData sound_data)
     {
+        sound_data->m_RefCount --;
+        if (sound_data->m_RefCount > 0)
+        {
+            return RESULT_OK;
+        }
```

Comment:
Since it _is_ a threaded system, I would rather like it for this alteration to be done _inside_ the mutex lock.

We might move it back out later, when it's clearer how this may be used from different threads (but then I suspect we need another spinlock or something)